homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,254 @@
+        status_sensor_available_value=0,
+        status_sensor_exists_values=[0, 1, 2],
+        value_key=Attribute.OUTDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE,
+        value_fn=lambda sensor, value: convert_temperature_if_needed(
+            sensor.unit_of_measurement, value
+        ),
+    ),
+    AprilaireSensorDescription(
+        key="dehumidification_status",
+        translation_key="dehumidification_status",
+        device_class=None,
+        state_class=None,
+        status_key=Attribute.DEHUMIDIFICATION_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1],
+        value_key=Attribute.DEHUMIDIFICATION_STATUS,
+        value_fn=lambda _, value: DEHUMIDIFICATION_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="humidification_status",
+        translation_key="humidification_status",
+        device_class=None,
+        state_class=None,
+        status_key=Attribute.HUMIDIFICATION_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1, 2],
+        value_key=Attribute.HUMIDIFICATION_STATUS,
+        value_fn=lambda _, value: HUMIDIFICATION_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="ventilation_status",
+        translation_key="ventilation_status",
+        status_key=Attribute.VENTILATION_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1],
+        value_key=Attribute.VENTILATION_STATUS,
+        value_fn=lambda _, value: VENTILATION_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="air_cleaning_status",
+        translation_key="air_cleaning_status",
+        status_key=Attribute.AIR_CLEANING_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1],
+        value_key=Attribute.AIR_CLEANING_STATUS,
+        value_fn=lambda _, value: AIR_CLEANING_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="fan_status",
+        translation_key="fan_status",
+        status_key=None,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[],
+        value_key=Attribute.FAN_STATUS,
+        value_fn=lambda _, value: FAN_STATUS_MAP.get(value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Aprilaire sensor devices."""
+
+    coordinator: AprilaireCoordinator = hass.data[DOMAIN][config_entry.unique_id]
+
+    assert config_entry.unique_id is not None
+
+    async_add_entities(
+        AprilaireSensor(coordinator, description, config_entry.unique_id)
+        for description in SENSOR_TYPES
+        if description.status_key is None
+        or description.status_sensor_exists_values is None
+        or coordinator.data.get(description.status_key)
+        in description.status_sensor_exists_values
```

Comment:
What do we do here?

Reply:
@joostlek There's not really a better way of writing this I don't think. It is getting all of the possible sensor types and including them if either

1. `status_key` or `status_sensor_exists_values` is `None`, indicating that the sensor should always be added
2. The coordinator data for the `status_key` is one of the allowed values in `status_sensor_exists_values`

The reason for the slightly clunky logic with the `status_sensor_exists_values` is that the thermostat returns a value for the status for which multiple may correspond to the sensor being available. For instance, the `Indoor temperature controlling sensor` should show in the UI if the `INDOOR_TEMPERATURE_CONTROLLING_SENSOR_STATUS` value is either 0 (normal) or 1/2 (installed but in an error state).